apis_core/generic/views.py

```diff
@@ -104,14 +105,35 @@ def get_table_class(self):
 
     def get_table_kwargs(self):
         kwargs = super().get_table_kwargs()
-        columns = self.request.GET.getlist("columns", [])
-        column_fields = [
-            field for field in self.model._meta.fields if field.name in columns
+
+        # we look at the selected columns and exclude
+        # all modelfields that are not part of that list
+        selected_columns = self.request.GET.getlist(
+            "columns",
+            self.get_filterset(self.get_filterset_class()).form["columns"].initial,
+        )
+        modelfields = self.model._meta.fields
+        kwargs["exclude"] = [
+            field.name for field in modelfields if field.name not in selected_columns
         ]
+
+        # now we look at the selected columns and
+        # add all modelfields and annotated fields that
+        # are part of the selected columns to the extra_columns
+        annotationfields = tuple()
+        for key, value in self.object_list.query.annotations.items():
+            fake_field = getattr(value, "field", value.output_field)
+            setattr(fake_field, "name", key)
+            annotationfields.append(fake_field)
+        extra_fields = list(
+            filter(lambda x: x.name in selected_columns, modelfields + annotationfields)
```

Comment:
```suggestion
            filter(lambda x: x.name in selected_columns, modelfields + tuple(annotationfields))
```
as we used list to append earlier, the list needs not be converted to a tuple for combining it with modelfields